src/actions.ts

```diff
@@ -270,9 +260,10 @@ export function activateByNid(nid:number|null, options?:{allowMove?: boolean, re
           SHARED.cm.scrollIntoView({top, bottom, left, right});
         }
         scroller.setAttribute('aria-activedescendent', newNode.element.id);
-        newNode.element.focus();
+        // if this timeout fires after the node has been torn down, don't focus
+        if(newNode.element) { newNode.element.focus(); }
```

Comment:
I just noticed that this `newNode.element` check is already happening on line 246. Unless the other code in the if block have side effects that would cause newNode.element to change, it should be safe to remove the if check here. But then there is an outstanding question of whether it is safe to access all the other stuff that might depend on dom nodes that have been removed. Will `SHARED.cm` still exist? Will `SHARED.cm.getScrollerElement()` still return an element?

Reply:
Ah, because if the ToggleEditor component is unmounted, `SHARED.cm` might get torn down! I'll include checks in the surrounding if-expressions.